### PR TITLE
Fix missing nodes in outline

### LINKED ISSUE #986: Outline is broken with nested items
For the following code: 

```zig
const Foo = struct {
    const Self = @This();
    pub fn foo() !Self {}
    const Bar = struct {};
};
```

the outline (`textDocument/documentSymbol`) contains neither `foo` function nor `Bar` struct. If I comment out either `Self` or `Bar` this  works though.

### PR DESCRIPTION
**Pull request by matklad**
Builds on top of #988.

Closes #986

### CODE REVIEW

File: src/analysis.zig [hunk 1]
```diff
@@ -1851,6 +1851,7 @@ fn addOutlineNodes(allocator: std.mem.Allocator, tree: Ast, child: Ast.Node.Inde
         .error_set_decl,
         => return,
         .container_decl,
+        .container_decl_trailing,
```
Comment (matklad) [inline src/analysis.zig:1854]: debating whether I should replace else bellow with an explicit list of tags

Comment (SuperAuguste) [reply src/analysis.zig:1854]: I don't think that's necessary here :)

**Comment by SuperAuguste** [comment]
Please rebase this and #988 :)

**Comment by matklad** [comment]
rebased!

**Review by SuperAuguste — APPROVED** [review]

**Pull request merged**